service/src/integrationTest/java/uk/gov/hmcts/reform/fpl/controllers/ApplicantAboutToStartControllerTest.java

```diff
@@ -26,8 +56,94 @@ void shouldPrepopulateApplicantDataWhenNoApplicantExists() {
             .data(Map.of("data", "some data"))
             .build();
 
+        given(authTokenGenerator.generate()).willReturn(serviceAuthToken);
+        given(organisationApi.findOrganisationById(userAuthToken, serviceAuthToken))
+            .willReturn(Organisation.builder().build());
+
         AboutToStartOrSubmitCallbackResponse callbackResponse = postAboutToStartEvent(caseDetails);
 
         assertThat(callbackResponse.getData()).containsKey("applicants");
     }
+
+    @BeforeEach
+    void setup() {
+        Organisation organisation = buildOrganisation();
+        given(authTokenGenerator.generate()).willReturn(serviceAuthToken);
+        given(organisationApi.findOrganisationById(userAuthToken, serviceAuthToken)).willReturn(organisation);
+    }
+
+    @Test
+    void shouldAddOrganisationDetailsToApplicantWhenOrganisationExists() {
+        CaseDetails caseDetails = CaseDetails.builder()
+            .data(Map.of("data", "some data"))
+            .build();
+
+        AboutToStartOrSubmitCallbackResponse callbackResponse = postAboutToStartEvent(caseDetails);
+
+        CaseData data = mapper.convertValue(callbackResponse.getData(), CaseData.class);
+
+        assertThat(data.getAllApplicants()).contains(buildApplicant());
+    }
+
+    @Test
+    void shouldFindOrganisation() {
+        Organisation organisation = buildOrganisation();
+        given(authTokenGenerator.generate()).willReturn(serviceAuthToken);
+        given(organisationApi.findOrganisationById(userAuthToken, serviceAuthToken)).willReturn(organisation);
+
+        Organisation actualOrganisation = organisationApi.findOrganisationById(userAuthToken, serviceAuthToken);
+
+        assertThat(actualOrganisation).isEqualTo(organisation);
+    }
+
+    @Test
+    public void shouldThrowExceptionWhenOrganisationNotFound() {
+        Exception exception = new FeignException.NotFound("", REQUEST, new byte[]{});
```

Comment:
hmm this test seems odd. You're throwing a specific type of feign exception but asserting a generic one, is this correct? 

I see no point extracting exception to a variable.

REQUEST isn't a very helpful variable name

Reply:
Removed exception variable and changed the assertion to specific feign exception.. 

not sure on changing REQUEST name, it's like this in other test classes. would you suggest something like REFERENCE_DATA_REQUEST instead? http request is to them..

Reply:
Resolving as don't think this is a big issue